lib/compress/zstd_opt.c

```diff
@@ -764,6 +764,152 @@ FORCE_INLINE_TEMPLATE U32 ZSTD_BtGetAllMatches (
     }
 }
 
+/*-*******************************
+*  LDM helper functions
+*********************************/
+
+/* Skips past srcSize bytes in an ldm seqstore */
+static void ldm_skipBytesInSeqStore(rawSeqStore_t* ldmSeqStore, size_t bytesToSkip) {
+    while (bytesToSkip > 0 && ldmSeqStore->pos < ldmSeqStore->size) {
+        rawSeq* seq = ldmSeqStore->seq + ldmSeqStore->pos;
+        if (bytesToSkip <= seq->litLength) {
+            /* Skip past srcSize literals */
+            seq->litLength -= (U32)bytesToSkip;
+            return;
+        }
+        bytesToSkip -= seq->litLength;
+        seq->litLength = 0;
+        if (bytesToSkip < seq->matchLength) {
+            seq->matchLength -= (U32)bytesToSkip;
```

Comment:
What happens when `matchLength < 3` after this line? That `matchLength` would be unrepresentable.

Reply:
I actually allow `matchLength < 3` since: we will always reject any `matchLength < MINMATCH` later on when we try to add LDMs, and I wanted to keep the functions as simple as possible to avoid any issues with miscounting the number of bytes we actually skipped. It would definitely make sense to note this specifically though.